default/ship_parts.txt

```diff
@@ -1134,11 +1134,22 @@ Count condition = And [
             ]
 '''
 
+// Beginning of Stacking macros
+/* For the AI and general uniformity, shields, detectors and cloaks are assigned a capacity.
+However, to prevent stacking, for each group X there is a X_STACKING macro which basically 
+first removes all effects of the used parts (by subtracting the summed capacity of all used parts)
```

Comment:
Aye, I think this came about after I accidentally designed a scout with two different scanning parts and got the worst of the two only, the real affect of the workaround was to make non-stacking parts work as expected and not semi-randomly. That it also makes use of capacity to assist the AI and the Pedia displays as well as let us script species traits more easily is a bonus (of sorts).

I would be very happy of the backend was rewritten to negate the need for workarounds but still let us have various parts and pick which should stack (eg Multi Spectral Shields and Trans Spatial Drive both give a stealth bonus, this lets us pick if they should stack or not whereas before we didn't have a huge amount of control without special exceptions in each part).